### ISSUE TITLE
Maximum update depth exceeded

### ISSUE THREAD

**Issue by sbuckpesch** [laurent22/joplin#8237]
Application content disappears out of a sudden (mostly when switching notes) and this error message appears:

## Error
Joplin encountered a fatal error and could not continue.

To report the error, please copy the *entire content* of this page and post it on Joplin forum or GitHub.

If the error persists you may try to [restart in safe mode](https://github.com/laurent22/joplin/issues/new?assignees=&labels=bug&projects=&template=bug_report.md&title=#), which will temporarily disable all plugins.

## Message
Maximum update depth exceeded. This can happen when a component repeatedly calls setState inside componentWillUpdate or componentDidUpdate. React limits the number of nested updates to prevent infinite loops.

## Version info
Joplin for Desktop

Copyright © 2016-2023 Laurent Cozic
Joplin 2.11.4 (prod, win32)

Client ID: a64ffe1c2d0248b7a788186240fedb47
Sync Version: 3
Profile Version: 42
Keychain Supported: Yes

Revision: 201c7d8

macOS theme: 1.3.13
Markdown Table: Sortable: 1.2.2
Outline: 1.5.9
Plugins
```
[
    {
        "id": "com.andrejilderda.macOSTheme",
        "name": "macOS theme",
        "enabled": true,
        "version": "1.3.13"
    },
    {
        "id": "outline",
        "name": "Outline",
        "enabled": true,
        "version": "1.5.9"
    },
    {
        "id": "com.hieuthi.joplin.markdown-table-sortable",
        "name": "Markdown Table: Sortable",
        "enabled": true,
        "version": "1.2.2"
    }
]
```

## Stack trace
```
Error: Maximum update depth exceeded. This can happen when a component repeatedly calls setState inside componentWillUpdate or componentDidUpdate. React limits the number of nested updates to prevent infinite loops.
    at checkForNestedUpdates (C:\Users\buckpesch\AppData\Local\Programs\Joplin\resources\app.asar\node_modules\react-dom\cjs\react-dom.development.js:27292:11)
    at scheduleUpdateOnFiber (C:\Users\buckpesch\AppData\Local\Programs\Joplin\resources\app.asar\node_modules\react-dom\cjs\react-dom.development.js:25475:3)
    at Object.enqueueSetState (C:\Users\buckpesch\AppData\Local\Programs\Joplin\resources\app.asar\node_modules\react-dom\cjs\react-dom.development.js:14067:7)
    at Component.setState (C:\Users\buckpesch\AppData\Local\Programs\Joplin\resources\app.asar\node_modules\react\cjs\react.development.js:354:16)
    at ItemList.updateStateItemIndexes (C:\Users\buckpesch\AppData\Local\Programs\Joplin\resources\app.asar\gui\ItemList.js:36:14)
    at ItemList.onScroll (C:\Users\buckpesch\AppData\Local\Programs\Joplin\resources\app.asar\gui\ItemList.js:70:14)
    at ItemList.componentDidUpdate (C:\Users\buckpesch\AppData\Local\Programs\Joplin\resources\app.asar\gui\ItemList.js:61:18)
    at commitLayoutEffectOnFiber (C:\Users\buckpesch\AppData\Local\Programs\Joplin\resources\app.asar\node_modules\react-dom\cjs\react-dom.development.js:23333:30)
    at commitLayoutMountEffects_complete (C:\Users\buckpesch\AppData\Local\Programs\Joplin\resources\app.asar\node_modules\react-dom\cjs\react-dom.development.js:24688:9)
    at commitLayoutEffects_begin (C:\Users\buckpesch\AppData\Local\Programs\Joplin\resources\app.asar\node_modules\react-dom\cjs\react-dom.development.js:24674:7)

Component stack

    at ItemList (C:\Users\buckpesch\AppData\Local\Programs\Joplin\resources\app.asar\gui\ItemList.js:8:9)
    at div
    at I (C:\Users\buckpesch\AppData\Local\Programs\Joplin\resources\app.asar\node_modules\styled-components\dist\styled-components.cjs.js:1:19385)
    at NoteListComponent (C:\Users\buckpesch\AppData\Local\Programs\Joplin\resources\app.asar\gui\NoteList\NoteList.js:51:88)
    at ConnectFunction (C:\Users\buckpesch\AppData\Local\Programs\Joplin\resources\app.asar\node_modules\react-redux\lib\components\connect.js:246:87)
    at div
    at I (C:\Users\buckpesch\AppData\Local\Programs\Joplin\resources\app.asar\node_modules\styled-components\dist\styled-components.cjs.js:1:19385)
    at NoteListWrapper (C:\Users\buckpesch\AppData\Local\Programs\Joplin\resources\app.asar\gui\NoteListWrapper\NoteListWrapper.js:16:49)
    at div
    at I (C:\Users\buckpesch\AppData\Local\Programs\Joplin\resources\app.asar\node_modules\styled-components\dist\styled-components.cjs.js:1:19385)
    at div
    at Resizable (C:\Users\buckpesch\AppData\Local\Programs\Joplin\resources\app.asar\node_modules\re-resizable\lib\index.es5.js:211:28)
    at div
    at Resizable (C:\Users\buckpesch\AppData\Local\Programs\Joplin\resources\app.asar\node_modules\re-resizable\lib\index.es5.js:211:28)
    at div
    at ResizableLayout (C:\Users\buckpesch\AppData\Local\Programs\Joplin\resources\app.asar\gui\ResizableLayout\ResizableLayout.js:46:45)
    at div
    at MainScreenComponent (C:\Users\buckpesch\AppData\Local\Programs\Joplin\resources\app.asar\gui\MainScreen\MainScreen.js:74:9)
    at ConnectFunction (C:\Users\buckpesch\AppData\Local\Programs\Joplin\resources\app.asar\node_modules\react-redux\lib\components\connect.js:246:87)
    at div
    at NavigatorComponent (C:\Users\buckpesch\AppData\Local\Programs\Joplin\resources\app.asar\gui\Navigator.js:7:1)
    at ConnectFunction (C:\Users\buckpesch\AppData\Local\Programs\Joplin\resources\app.asar\node_modules\react-redux\lib\components\connect.js:246:87)
    at exports.ThemeProvider (C:\Users\buckpesch\AppData\Local\Programs\Joplin\resources\app.asar\node_modules\styled-components\dist\styled-components.cjs.js:1:25117)
    at ae (C:\Users\buckpesch\AppData\Local\Programs\Joplin\resources\app.asar\node_modules\styled-components\dist\styled-components.cjs.js:1:13440)
    at RootComponent (C:\Users\buckpesch\AppData\Local\Programs\Joplin\resources\app.asar\gui\Root.js:103:1)
    at ConnectFunction (C:\Users\buckpesch\AppData\Local\Programs\Joplin\resources\app.asar\node_modules\react-redux\lib\components\connect.js:246:87)
    at ErrorBoundary (C:\Users\buckpesch\AppData\Local\Programs\Joplin\resources\app.asar\gui\ErrorBoundary.js:21:9)
    at Provider (C:\Users\buckpesch\AppData\Local\Programs\Joplin\resources\app.asar\node_modules\react-redux\lib\components\Provider.js:19:3)
```

## Describe what you expected to happen
Just the application to keep working.

**Comment by Daeraxa** [comment]
Does this happen in safe mode? Have you tried disabling any of the plugins?

**Comment by sbuckpesch** [comment]
I did not activate the safe mode yet. I disabled the "Markdown Table: Sortable" plugin for now, as this might cause re-rendering issues as the pages causing the crash contain multiple tables.
If the problem occurs again, I will come back here and post more detailled debug information...

**Comment by laurent22** [comment]
I cannot replicate the bug but based on the stacktrace, it should be fixed in 659f602a125d91ff97388f82c6300dbecc89376c
